src/state/utils/store_utils.ts

```diff
@@ -32,3 +32,33 @@ export const addDerivedStateSubscriber = <T, const U>(
     }
   );
 };
+
+/**
+ * Returns a new callback function that wraps the original callback, but debounces
+ * repeated calls by a specified number of milliseconds.
+ * @param callback The original callback function to debounce.
+ * @param debounceMs The number of milliseconds to wait before calling the
+ * debounced callback. Defaults to 250ms.
+ */
+export const makeDebouncedCallback = <T, U, CallbackFn extends (state: T) => Partial<U> | undefined | void>(
+  callback: CallbackFn,
+  debounceMs: number = 250
```

Comment:
Out of curiosity, how do you choose an amount of time for a debounced callback?

Reply:
It's arbitrary! I might actually tweak this to 100ms because that's what it was originally.
The goal is to make sure that we're not firing a ton of expensive computations while a user is rapidly flipping through values. Since the time interval is a slider, it'll fire callbacks on every value in a continuous range.